### ISSUE TITLE
Type of class variable  from "grandparent" is replaced by untyped-subclass

### ISSUE THREAD

**Issue by ashb** [python/mypy#6511]
Minimal repro case:

```python
from typing import Iterable

class A:
    a: Iterable[str] = ("a", )

class B(A):
    a = ("a", "b")

class C(B):
    a = ("a", "b", "c")
```

And the error:

> ```a.py:10: error: Incompatible types in assignment (expression has type "Tuple[str, str, str]", base class "B" defined the type as "Tuple[str, str]")```

The error is reported on the assignment in `C`.

I would have expected that B.a would have the same type as A.a since B.a doesn't have an explicit type.

Maybe relates to #4547 or #2510, or maybe I'm just expecting mypy to do something that it won't :)

**Comment by ilevkivskyi** [comment]
I think this is essentially a duplicate of https://github.com/python/mypy/issues/4547

**Comment by ashb** [comment]
Could be yes - I'm not familiar enough with Mypy to say, and I thought it might have been specific to having three classes in the hierarchy, and I got a different error message.
